Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/SpringForceField.inl

```diff
@@ -104,6 +104,74 @@ void SpringForceField<DataTypes>::updateTopologyIndicesFromSprings()
     areSpringIndicesDirty = false;
 }
 
+template <class DataTypes>
+void SpringForceField<DataTypes>::applyRemovedEdges(const sofa::core::topology::EdgesRemoved* edgesRemoved, sofa::Index mstateId)
+{
+    if (edgesRemoved == nullptr)
+        return;
+
+    const type::vector<sofa::core::topology::Topology::EdgeID>& edges = edgesRemoved->getArray();
+
+    if (edges.empty())
+        return;
+    
+    core::topology::BaseMeshTopology* modifiedTopology;
+    if (mstateId == 0)
+    {
+        modifiedTopology = this->getMState1()->getContext()->getMeshTopology();
+    }
+    else
+    {
+        modifiedTopology = this->getMState2()->getContext()->getMeshTopology();
+    }
+
+    if (modifiedTopology == nullptr)
+        return;
+
+    type::vector<Spring>& springsValue = *sofa::helper::getWriteAccessor(this->springs);
+    
+    const auto& topologyEdges = modifiedTopology->getEdges();
+
+    type::vector<sofa::Index> springIdsToDelete;
+
+    for (const auto& edgeId : edges) // iterate on the edgeIds to remove and save the respective point pairs
+    {
+        auto& firstPointId = topologyEdges[edgeId][0];
+        auto& secondPointId = topologyEdges[edgeId][1];
+
+        // sane default value to check against, if no spring is found for the edge
+        int springIdToDelete = -1;
+        sofa::Index i = 0;
+
+        for (const auto& spring : springsValue) // loop on the list of springs to find the spring with targeted pointIds
+        {
+            auto& firstSpringPointId = mstateId == 0 ? spring.m1 : spring.m2;
+            auto& secondSpringPointId = mstateId == 0 ? spring.m2 : spring.m1;
+
+            if (firstSpringPointId == firstPointId && secondSpringPointId == secondPointId)
+            {
+                dmsg_info() << "Spring " << spring << " has an edge to be removed: REMOVED edgeId: " << edgeId;
+                springIdToDelete = i;
+                break; // break as soon as the first matching spring is found. TODO is there a valid case for having multiple springs on the same topology edge?
```

Comment:
I think we check in AddSpring to avoid redundant springs... but I'm not 100% sure.